src/transformers/models/gemma/modeling_gemma.py

```diff
@@ -1218,6 +1231,7 @@ def prepare_inputs_for_generation(
                 "past_key_values": past_key_values,
                 "use_cache": use_cache,
                 "attention_mask": attention_mask,
+                "_length": int(cache_position[-1]) + 1,
```

Comment:
This is redundant of  `cache_position`, however, this is the only way I can figure out to make the dynamic length computation works with `torch.compile`.